native/logik_simulation/src/data/subnet.rs

```diff
@@ -15,17 +17,50 @@ impl Subnet {
         self.state
     }
     
-    pub(crate) fn update(&mut self, state: SubnetState) {
-        self.state = state;
+    pub(crate) fn update(&mut self, state: SubnetState) -> bool {
+        if self.state == state {
+            false
+        } else {
+            self.state = state;
+            true
+        }
     }
 }
 
 /// An enum to represent the different states that a subnet can have
 #[repr(u8)]
-#[derive(Debug, Eq, PartialEq, Clone, Copy)]
+#[derive(Debug, Eq, PartialEq, Clone, Copy, Hash)]
 pub enum SubnetState {
     Floating = 0,
     Off = 1,
     On = 2,
     Error = 3,
+}
+
+impl SubnetState {
+    pub(crate) fn work_out_diff(diff: &HashSet<SubnetState>) -> SubnetState {
+        let filtered = diff.into_iter()
+            .filter(|e| e != &&SubnetState::Floating)
+            .collect::<Vec<_>>();
+        
+        match filtered.len() {
+            0 => SubnetState::Floating,
+            1 => *filtered[0],
+            _ => SubnetState::Error,
+        }
```

Comment:
Will this not fail when there is a floating and a non-zero state on a subnet?

Reply:
No, because we filter out the floating state when considering how many different states there are.

Reply:
Ok then I understand. It might be worth putting a comment about that there.